src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -285,12 +285,8 @@ private ReadOnlyMemory<byte> ReadChunkFromConnectionBuffer(int maxBytesToRead, C
 
                             while (true)
                             {
-                                // Folded header was deprecated in RFC 7230 3.2.4. Thus we don't support it.
                                 _connection._allowedReadLineBytes = MaxTrailingHeaderLength;
-                                if (!_connection.TryReadNextLine(out currentLine))
-                                {
-                                    break;
-                                }
+                                currentLine = _connection.ReadNextResponseHeaderLineAsync(foldedHeadersAllowed: true).Result;
```

Comment:
This should use await and .ConfigureAwait(false) etc.

Reply:
I didn't use `await` at the first place because I want to preserve the current synchronous method signature.

I think it's fine to make this method `async` since it will be called from async context. Will change.

Reply:
> want to preserve the current synchronous method signature.

I've not looked to see what location you're referring to, but we should not be blocking anywhere in this codebase on a task.

Reply:
Support folded header is harder than I expect:
1. We need to asynchronously read/buffer additional information for folded headers,
2. Current function is in synchronous context and will be called from synchronous context, which uses lots of `Span` related APIs and existing functions (`Span` cannot be declared in async methods as it uses stack space).

Given my time frame, I will open a new issue for folded trailing header support instead of adding it in this PR.